week06_policy_based/atari_wrappers.py

```diff
@@ -274,53 +320,24 @@ def reset(self, **kwargs):
         return self.env.reset(**kwargs)
 
 
-class TFSummaries(SummariesBase):
-    """ Writes env summaries using TensorFlow."""
-
-    def __init__(self, env, prefix=None, running_mean_size=100, step_var=None):
-
-        super().__init__(env, prefix, running_mean_size)
-
-        raise NotImplementedError(
-            'TFSummaries class no longer works with the latest Tensorflow because '
-            'tf.train.get_global_step() was removed. Use NumpySummaries instead.')
-
-        import tensorflow as tf
-        self.step_var = (step_var if step_var is not None
-                         else tf.train.get_global_step())
+# magic for parallel launching of environments
+class _thunk:
+    def __init__(self, i, env_id, **kwargs):
+        self.env_id = env_id
+        self.i = i
+        self.kwargs = kwargs
 
-    def add_summary_scalar(self, name, value):
-        import tensorflow as tf
-        tf.contrib.summary.scalar(name, value, step = self.step_var)
+    def __call__(self):
+        return nature_dqn_env(
+            self.env_id,
+            summaries=False,
+            clip_reward=False,
+            **self.kwargs,
+        )
 
 
-class NumpySummaries(SummariesBase):
```

Comment:
I don't think we should remove `NumpySummaries`.
Many students still prefer to use it.
I think we should keep the ability to choose the type of summaries via argument passed to `nature_dqn_env()`.

Reply:
Thank you for your comment!
I guess the students still use it due to the lack of alternatives since `TFSummaries` has been deprecated.

Anyway, I've returned `SummariesBase` and `NumpySummaries` and replaced `TFSummaries` with `TensorboardSummaries`.

Reply:
Thank you!